dali/operators/decoder/audio/generic_decoder.cc

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/decoder/audio/generic_decoder.h"
+
+namespace dali {
+
+namespace {
+
+template<typename SampleType>
+size_t read_samples(SNDFILE *snd_file, span<SampleType> output) {
```

Comment:
I'd leave it undefined so that it doesn't even compile for not supported types (fewer surprises at runtime)

Reply:
Done